cmd/soroban-rpc/internal/test/archive_test.go

```diff
@@ -0,0 +1,25 @@
+package test
+
+import (
+	"net/http"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestArchiveUserAgent(t *testing.T) {
+	userAgents := sync.Map{}
+	cfg := &Test{
+		historyArchiveProxyCallback: func(r *http.Request) {
+			userAgents.Store(r.Header["User-Agent"][0], "")
+		},
+	}
+	NewTest(t, cfg)
```

Comment:
it seems unintuitive for NewTest() to both take in a *Test parameter and return a *Test value. I think the functional options pattern could be a good fit here:

https://github.com/tmrts/go-patterns/blob/master/idiom/functional-options.md

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
yes, great point, i went with similar pattern as horizon internal integration tests, using a TestConfig to encapsulate, thanks! https://github.com/stellar/soroban-rpc/pull/56/commits/cdf9481bef898ce8b878400b2901b9fcabb02980